org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/UriExtensionsTest.java

```diff
@@ -117,6 +119,20 @@ public void test_toUri_14() throws Exception {
 				uriExtensions.toUri("file:///c%3A/Users/dietrich/%D0%A4%D1%83%20%D0%91%D0%B0%D1%80.mydsl"));
 	}
 
+	@Test
+	public void test_toUri_15() throws Exception {
+		String uri = "something:/path/to/res%20ource";
+		String encodedUri = URLEncoder.encode(uri, StandardCharsets.UTF_8.toString());
+		assertEquals(URI.createURI(uri), uriExtensions.toUri(encodedUri));
+	}
+
+	@Test
+	public void test_toUri_16() throws Exception {
+		String uri = "something:/path/to/res#ource";
+		String encodedUri = URLEncoder.encode(uri, StandardCharsets.UTF_8.toString());
+		assertEquals(URI.createURI(uri, true, URI.FRAGMENT_NONE), uriExtensions.toUri(encodedUri));
```

Comment:
the intent of this is to test that a file with `#`, when converted to the EMF URI on the server, does not end up being `res` with fragment `ource`.